server.go

```diff
@@ -528,6 +528,13 @@ func (s *Server) DatabaseExists(name string) bool {
 	return s.databases[name] != nil
 }
 
+// Database returns a database given a database name.
+func (s *Server) database(name string) *database {
```

Comment:
don't think there's a need for this. Since the databases don't have mutexes in them, it's up to the server to lock all this down.